src/MagnumPlugins/KtxImporter/KtxHeader.h

```diff
@@ -0,0 +1,458 @@
+#ifndef Magnum_Trade_KtxHeader_h
+#define Magnum_Trade_KtxHeader_h
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019,
+                2020, 2021 Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2021 Pablo Escobar <mail@rvrs.in>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Vector3.h>
+
+/* Used by both KtxImporter and KtxImageConverter, which is why it isn't
+   directly inside KtxImporter.cpp. OTOH it doesn't need to be exposed
+   publicly, which is why it has no docblocks. */
+
+namespace Magnum { namespace Trade { namespace Implementation {
+
+/* Taken from magnum/src/MagnumExternal/Vulkan/flextVk.h
+   (commit 9d4a8b49943a084cff64550792bb2eba223e0e03)
+
+   Contains all formats from:
+   - 1.2 core (without KHR aliases)
+   - EXT_texture_compression_astc_hdr
+   - IMG_format_pvrtc */
+enum VkFormat : UnsignedInt {
+    VK_FORMAT_UNDEFINED = 0,
```

Comment:
Just a random though I had today -- since we don't really need to use the enum values anywhere, what if `formatMapping.hpp` had directly the numeric values, with the enum string perhaps only as a documentation comment? Since you're extracting the data from one enum already, why not go all the way and dereference it fully. That way updating for newly added formats would be just rerunning the `.py` script, without also having to copypaste this enum here.

Reply:
The importer uses the original Vulkan enums to detect swizzled BGR/BGRA formats that don't exist in `PixelFormat`, so the enum is still needed. I could try to extract the enum from flextVk.h? Alternative B: a custom flextVk template, but that might go out of sync too. The last option would be amazing if we had the .xml that Khronos use to generate the vulkan_core.h used by dfdutils that has some of the unreleased (but supported by KTX) formats like 3D ASTC.

Reply:
> swizzled BGR/BGRA formats

Right. What about having this VkFormat enum list only the formats we need to act on in some way, and the rest being encoded directly as numeric values in `formatMapping.hpp` (so extracted from both `flextVk.h` and `Vk/PixelFormat.h`)? That way updating would in most cases mean regenerating only the `formatMapping.hpp`, and the `VkFormat` enum will get extended only in rare scenarios.

> 3D ASTC

Completely forgot about these, sigh. [And seems like I asked already](https://github.com/KhronosGroup/KTX-Specification/pull/97#pullrequestreview-293626218), two years ago -- no way to get access to that XML I guess :(